Mobile deployment docs overhaul (#4606)
mattermost/docs: source/mobile/deploy-mobile-apps-using-emm-provider.rst

Deploying Mobile Apps Using an EMM Provider
===========================================

An EMM Provider builds `Enterprise Mobile Management software <https://en.wikipedia.org/wiki/Enterprise_mobility_management>`__ that helps enterprise teams manage secure mobile endpoints with managed app configuration. 

You can use an EMM to: 

> - Enforce users to download a mobile app.

Author: Would this read better as: "Control which mobile app users download."?

Author: @esethna - which statement is closer to the user's reality when using EMM providers?

Reviewer: it should be directly associated with MM, something like

```
Established the Mattermost pre-built or custom apps to be managed by your organization
```

or something along those lines

Manage App Configuration Using AppConfig
----------------------------------------

AppConfig is our recommended approach for app configuration and management. It was introduced by the `AppConfig Community <https://www.appconfig.org/about/>`__, a group of leading EMM providers and app developers who have come together to make it easier for developers and customers to drive mobility in business. 

AppConfig provides an easy way to configure enterprise mobile apps with any of the EMM providers listed on the `AppConfig website <https://www.appconfig.org/members/>`__. Using AppConfig, you can manage default settings and security controls on both the public app stores, and custom-built mobile clients. For example, you can pre-configure your Mattermost server URL and username.

See our `Mattermost AppConfig Values <https://docs.mattermost.com/mobile/mobile-appconfig.html#mattermost-appconfig-values>`__ documentation for details on the configuration options that can be sent from the EMM provider to Mattermost Mobile apps. 

.. important::

>     Mattermost only supports the AppConfig standard for securing Mattermost Mobile apps via an EMM provider due to incompatibilities with app wrapping and React Native applications. React Native is the technology used to develop the Mattermost Mobile apps. Different EMM vendors refer to “wrapping” in different ways, but it ultimately comes down to unpacking the mobile client bundle, injecting additional SDKs, and re-packaging/re-signing. 

Reviewer 2: ```suggestion
    
    Mattermost only supports the AppConfig standard for securing Mattermost Mobile apps via an EMM provider due to incompatibilities with app wrapping and React Native applications. React Native is the technology used to develop the Mattermost Mobile apps. Different EMM vendors refer to “wrapping” in different ways, but it ultimately comes down to unpacking the mobile client bundle, injecting additional SDKs, and re-packaging/re-signing. 
```

Host Your Own Push Proxy Service
--------------------------------

A push proxy is a key technology behind notification transmission. It enables notifications between the server and a Mobile app. Mattermost provides a self-hosted push proxy you can deploy called the `Mattermost Push Notification Service (MPNS) <https://docs.mattermost.com/deployment/deployment.html#push-notification-service>`__. However, if you choose to build custom versions of the Mattermost Mobile app, you must host your own instance of the MPNS by compiling your own MPNS from the `open source repository <https://github.com/mattermost/mattermost-push-proxy>`__, or by using the `pre-compiled version available on GitHub <https://github.com/mattermost/mattermost-push-proxy/releases>`__. 

See our `developer documentation <https://developers.mattermost.com/contribute/mobile/push-notifications/service/>`__ on installing the Mattermost Push Notification Service for details.

.. note::

- The MPNS should be behind your firewall inside your private network, or in your DMZ, in a way that the Mattermost server can access it.

> - The MPNS does not connect with Mobile apps directly; it parses and forwards push notifications from the Mattermost Server to the Apple Push Notification Service (APNS) or the Firebase Cloud Messaging (FCM).

Reviewer 2: ```suggestion
- The MPNS does not connect with Mobile apps directly; it parses and forwards push notifications from the Mattermost server to the Apple Push Notification Service (APNS) or the Firebase Cloud Messaging (FCM).
```

  
Securing Push Notifications
~~~~~~~~~~~~~~~~~~~~~~~~~~~

To secure your push notifications, ensure you use encrypted TLS connections between:

- MPNS and Apple Push Notification Service
- MPNS and Google’s Firebase Cloud Messaging
- MPNS and your Mattermost server

Enabling MPNS in Mattermost
~~~~~~~~~~~~~~~~~~~~~~~~~~~

1. Go to **System Console > Environment > Push Notification Server**.
2. Under **Enable Push Notifications**, select **Manually enter Push Notification Service location**.
3. Enter the location of your MPNS in the **Push Notification Server**  field, then select **Save**.
4. (Optional) Customize mobile push notification contents. Most deployments choose to include the full message content sent in the notification payload.
    i. Go to **System Console > Site Configuration > Notifications**.
    ii. Under **Push Notification Contents**, select the type of information to include in push notifications, then select **Save**.
5. Subscribe to `Mattermost Security Bulletins <https://mattermost.com/security-updates/#sign-up>`__. When notified of security updates, apply them promptly.

.. note:: 

>     As part of the process of building the applications you will need to sign the applications. You must also obtain the appropriate certificate for both Android and iOS. If this isn't done, the applications won't be able to interact with your instance of the MPNS. Once this is complete, you can proceed with the deployment of your MPNS instance.

Reviewer 2: ```suggestion
    As part of the process of building the applications you'll need to sign the applications. You must also obtain the appropriate certificate for both Android and iOS. If this isn't done, the applications won't be able to interact with your instance of the MPNS. Once this is complete, you can proceed with the deployment of your MPNS instance.
```

>     As part of the process of building the applications you will need to sign the applications. You must also obtain the appropriate certificate for both Android and iOS. If this isn't done, the applications won't be able to interact with your instance of the MPNS. Once this is complete, you can proceed with the deployment of your MPNS instance.

Reviewer 2: I think this is slightly too right-indented but it doesn't seem to negatively affect the preview render.

We also recommend following our `recommended steps to secure your deployment <https://docs.mattermost.com/mobile/mobile-appstore-install.html>`__ and to review the following commonly-asked questions about data security on mobile devices:

- `How data is handled on a device after an account is deleted? <https://docs.mattermost.com/mobile/mobile-faq.html#how-is-data-handled-on-mobile-devices-after-a-user-account-is-deactivated>`__
- `What post metadata is sent in mobile push notifications? <https://docs.mattermost.com/mobile/mobile-faq.html#what-post-metadata-is-sent-in-mobile-push-notifications>`__
- `What are my options for securing the Mobile apps? <https://docs.mattermost.com/mobile/mobile-faq.html#what-are-my-options-for-securing-the-mobile-apps>`__
- `What are my options for securing push notifications? <https://docs.mattermost.com/mobile/mobile-faq.html#what-are-my-options-for-securing-push-notifications>`__

Per-app VPN
~~~~~~~~~~~

A common approach is to use a per-app VPN. This provides a connection to the VPN when needed (on-demand). If using a per-app VPN with Mattermost, you can configure the following options:

> - **useVPN**: Mattermost waits until the connection to the VPN server is established before making any requests (otherwise they will fail). Only supported on iOS given Android OS cannot support waiting - still works but the first connection attempt may fail. 

Reviewer 2: ```suggestion
- **useVPN:** Mattermost waits until the connection to the VPN server is established before making any requests (otherwise they will fail). This is only supported on iOS as Android OS cannot support waiting. It still works but the first connection attempt may fail.
```

> - **timeoutVPN** (iOS only): How long to wait for the connection to the VPN server before trying.

Reviewer 2: ```suggestion
- **timeoutVPN (iOS only):** How long to wait for the connection to the VPN server before trying.
```

Deploying Mobile Apps Using an EMM Provider
===========================================

> An EMM Provider builds `Enterprise Mobile Management software <https://en.wikipedia.org/wiki/Enterprise_mobility_management>`__ that helps enterprise teams manage secure mobile endpoints with managed app configuration. 

Reviewer: ```suggestion
An EMM Provider `Enterprise Mobile Management software <https://en.wikipedia.org/wiki/Enterprise_mobility_management>`__ that helps enterprise teams manage secure mobile endpoints with managed app configuration. 
```

Author: Rewrote this sentence for clarity. Thanks!

You can use an EMM to: 

- Enforce users to download a mobile app.

> - Set default server connections.

Reviewer: ```suggestion
- Set default server url address.
```

- Restrict users from changing servers.
- Enforce security policies.

An EMM provider pushes Mattermost Mobile apps to EMM-enrolled devices. This approach is recommended for organizations that typically use EMM solutions to deploy Mobile apps to meet security and compliance policies. 

Manage App Configuration Using AppConfig
----------------------------------------

AppConfig is our recommended approach for app configuration and management. It was introduced by the `AppConfig Community <https://www.appconfig.org/about/>`__, a group of leading EMM providers and app developers who have come together to make it easier for developers and customers to drive mobility in business. 

AppConfig provides an easy way to configure enterprise mobile apps with any of the EMM providers listed on the `AppConfig website <https://www.appconfig.org/members/>`__. Using AppConfig, you can manage default settings and security controls on both the public app stores, and custom-built mobile clients. For example, you can pre-configure your Mattermost server URL and username.

> See our `Mattermost AppConfig Values <https://docs.mattermost.com/mobile/mobile-appconfig.html#mattermost-appconfig-values>`__ documentation for details on the configuration options that can be sent from the EMM provider to Mattermost Mobile apps. 

Author: This link appears to be working. Will confirm after merge.

.. important::
    
    Mattermost only supports the AppConfig standard for securing Mattermost Mobile apps via an EMM provider due to incompatibilities with app wrapping and React Native applications. React Native is the technology used to develop the Mattermost Mobile apps. Different EMM vendors refer to “wrapping” in different ways, but it ultimately comes down to unpacking the mobile client bundle, injecting additional SDKs, and re-packaging/re-signing. 

    Mattermost doesn’t support app wrapping, and Mattermost Mobile apps won't function properly when using app wrapping (e.g., Websockets for real-time messaging will break). Use app wrapping/containerization technology at your own risk.

> Connecting to your Private Network Mattermost Instance

Author: @enahum - Are you able to assist with this? ^

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

You need to set up a way to connect to your private network Mattermost instance, using an external proxy with encrypted transport through HTTPS and WSS network connections.

Depending on your security policies, we recommend deploying Mattermost behind a VPN and using a per-app VPN with your EMM provider, or a mobile VPN client.

Also consider deploying a mobile VPN client with multi-factor authentication (MFA) to your preferred login method, such as GitLab SSO with MFA, or run Mattermost Enterprise Edition with MFA.

> Host Your Own Push Proxy Service

Author: The Host Your Own Push Proxy Service details have been moved to the Mobile Push Notifications page, where all push notification services are now documented. Thanks!

1. Go to **System Console > Environment > Push Notification Server**.
2. Under **Enable Push Notifications**, select **Manually enter Push Notification Service location**.
3. Enter the location of your MPNS in the **Push Notification Server**  field, then select **Save**.
4. (Optional) Customize mobile push notification contents. Most deployments choose to include the full message content sent in the notification payload.
    i. Go to **System Console > Site Configuration > Notifications**.
    ii. Under **Push Notification Contents**, select the type of information to include in push notifications, then select **Save**.
5. Subscribe to `Mattermost Security Bulletins <https://mattermost.com/security-updates/#sign-up>`__. When notified of security updates, apply them promptly.

.. note:: 

    As part of the process of building the applications you will need to sign the applications. You must also obtain the appropriate certificate for both Android and iOS. If this isn't done, the applications won't be able to interact with your instance of the MPNS. Once this is complete, you can proceed with the deployment of your MPNS instance.

Enroll Devices
--------------

> When building your own custom versions of Mattermost Mobile apps, consider your organization’s mobile policy:

Reviewer: ```suggestion
When building your own custom versions or deploying the pre-built Mattermost Mobile apps, consider your organization’s mobile policy:
```
